homeassistant/components/config/scene.py

```diff
@@ -73,4 +59,15 @@ def _write_value(self, hass, data, config_key, new_value):
         # supporting more fields in the future.
         updated_value.update(new_value)
 
-        data[index] = updated_value
+        for index, cur_value in enumerate(data):
+            # When people copy paste their scenes to the config file,
+            # they sometimes forget to add IDs. Fix it here.
+            if CONF_ID not in cur_value:
+                cur_value[CONF_ID] = uuid.uuid4().hex
+
+            elif cur_value[CONF_ID] == config_key:
+                data[index] = updated_value
+                break
```

Comment:
Because we break here, when we update a scene we will never fix all IDs after the scene that we are updating, but this was a pre-existing thing before this PR.